src/snowcli/cli/common/snow_cli_typer.py

```diff
@@ -0,0 +1,227 @@
+
+
+def _are_tracebacks_enabled() -> bool:
+    return (
+        snow_cli_global_context_holder.get_global_context_copy().logging_and_exception_handling.enable_tracebacks
+    )
+
+
+def _with_exception_handling(f, *args, **kwargs):
+    try:
+        f(*args, **kwargs)
+    except ClickException as exception:
+        raise exception from None
+    except Exception as exception:
+        if _are_tracebacks_enabled():
+            raise exception from None
+        else:
+            raise ClickException(
+                "An unexpected exception occurred:\n"
+                + f"{exception.__class__.__name__}: {exception}\n"
+                + "Use --debug option to see the full traceback."
+            ) from None
+
+
+class SnowCliTyper(Typer):
+    def __init__(
+        self,
+        *,
+        name: Optional[str] = Default(None),
+        cls: Optional[Type[TyperGroup]] = Default(None),
+        invoke_without_command: bool = Default(False),
+        no_args_is_help: bool = Default(False),
+        subcommand_metavar: Optional[str] = Default(None),
+        chain: bool = Default(False),
+        result_callback: Optional[Callable[..., Any]] = Default(None),
+        # Command
+        context_settings: Optional[Dict[Any, Any]] = Default(None),
+        callback: Optional[Callable[..., Any]] = Default(None),
+        help: Optional[str] = Default(None),
+        epilog: Optional[str] = Default(None),
+        short_help: Optional[str] = Default(None),
+        options_metavar: str = Default("[OPTIONS]"),
+        add_help_option: bool = Default(True),
+        hidden: bool = Default(False),
+        deprecated: bool = Default(False),
+        add_completion: bool = True,
+        # Rich settings
+        rich_markup_mode: MarkupMode = None,
+        rich_help_panel: Union[str, None] = Default(None),
+        pretty_exceptions_enable: bool = True,
+        pretty_exceptions_show_locals: bool = True,
+        pretty_exceptions_short: bool = True,
+    ):
+        super().__init__(
+            name=name,
+            cls=cls,
+            invoke_without_command=invoke_without_command,
+            no_args_is_help=no_args_is_help,
+            subcommand_metavar=subcommand_metavar,
+            chain=chain,
+            result_callback=result_callback,
+            context_settings=context_settings,
+            callback=callback,
+            help=help,
+            epilog=epilog,
+            short_help=short_help,
+            options_metavar=options_metavar,
+            add_help_option=add_help_option,
+            hidden=hidden,
+            deprecated=deprecated,
+            add_completion=add_completion,
+            rich_markup_mode=rich_markup_mode,
+            rich_help_panel=rich_help_panel,
+            pretty_exceptions_enable=pretty_exceptions_enable,
+            pretty_exceptions_show_locals=pretty_exceptions_show_locals,
+            pretty_exceptions_short=pretty_exceptions_short,
+        )
+
+    def callback(
```

Comment:
`I would consider using *args, **kwargs instead of all parameters to indicate that we simply pass them to super method.`

I thought about it but doesn't it break type hints?

Reply:
`Do we need all methods?`

I overridden 3 methods which in my opinion we need to override.
* command - to handle exceptions in commands
* callback - to handle exceptions before commands (in callbacks)
* add_typer - to provide common logic (including exception handling) to all typer instances (command subgroups) and forbid using simple `Typer`

Reply:
`How we plan to maintain compatibility with typer/click?`

Do you think adjusting our interface on Typer's interface change will be a problem? I hope it won't happen often.

Reply:
We agreed to not use the idea of overriding Typer's interface as it can become problematic in the future and as the result we returned to previous solution: https://github.com/Snowflake-Labs/snowcli/pull/192